blockchain/chain_sync/src/sync.rs

```diff
@@ -134,12 +140,39 @@ impl<DB> ChainSyncer<DB>
 where
     DB: BlockStore,
 {
-    /// Starts syncing process
-    pub async fn sync(mut self) -> Result<(), Error> {
+    pub async fn start(mut self) -> Result<(), Error> {
         self.net_handler.spawn(Arc::clone(&self.peer_manager));
 
-        info!("Bootstrapping peers to sync");
+        while let Some(event) = self.network.receiver.next().await {
+            if let NetworkEvent::Hello { source, message } = &event {
+                if let Ok(fts) = self
+                    .fetch_tipset(
+                        source.clone(),
+                        &TipSetKeys::new(message.heaviest_tip_set.clone()),
+                        1,
+                    )
+                    .await
+                {
+                    if self.inform_new_head(&source.clone(), &fts).await.is_err() {
+                        warn!(
+                            "failed to sync with provided tipset cid: {:?}",
+                            message.heaviest_tip_set.clone()
```

Comment:
Should never need to clone to log
```suggestion
                            &message.heaviest_tip_set
```